iothub/device/src/Transport/Mqtt/MqttIotHubAdapter.cs

```diff
@@ -719,6 +719,12 @@ private async void ProcessMessageAsync(IChannelHandlerContext context, Packet pa
 
                     case PacketType.PUBLISH:
                         ProcessPublish(context, (PublishPacket)packet);
+                        // Checking if acknowledgement needs to be sent after receiving method from Hub with QoS = 1
```

Comment:
You need to make it more clear here why we are only sending the ack if it was a direct method message with QoS 1. Explain that the other QoS messages are ack'd elsewhere